### ISSUE TITLE
Kafka Connector Sink: breaking change with 1.6.3 ?

### ISSUE THREAD

**Issue by vdesabou** [snowflakedb/snowflake-kafka-connector#369]
I have this [example](https://github.com/vdesabou/kafka-docker-playground/tree/master/connect/connect-snowflake-sink) which is working perfectly fine with versions before `1.6.3` but with `1.6.3` , the *exact* same example is getting following error:

```
[SF_KAFKA_CONNECTOR] Establishing a JDBC connection with url:jdbc:snowflake://xxxxxx.snowflakecomputing.com:443 (com.snowflake.kafka.connector.internal.SnowflakeConnectionServiceV1:63)
[2021-12-20 15:18:43,058] ERROR Validate: Error connecting to snowflake:
[SF_KAFKA_CONNECTOR] Exception: Failed to connect to Snowflake Server
[SF_KAFKA_CONNECTOR] Error Code: 1001
[SF_KAFKA_CONNECTOR] Detail: Snowflake connection issue, reported by Snowflake JDBC
[SF_KAFKA_CONNECTOR] Message: Incorrect username or password was specified.
[SF_KAFKA_CONNECTOR] net.snowflake.client.core.SessionUtil.newSession(SessionUtil.java:615)
[SF_KAFKA_CONNECTOR] net.snowflake.client.core.SessionUtil.openSession(SessionUtil.java:285)
[SF_KAFKA_CONNECTOR] net.snowflake.client.core.SFSession.open(SFSession.java:446)
[SF_KAFKA_CONNECTOR] net.snowflake.client.jdbc.DefaultSFConnectionHandler.initialize(DefaultSFConnectionHandler.java:104)
[SF_KAFKA_CONNECTOR] net.snowflake.client.jdbc.DefaultSFConnectionHandler.initializeConnection(DefaultSFConnectionHandler.java:79)
[SF_KAFKA_CONNECTOR] net.snowflake.client.jdbc.SnowflakeConnectionV1.initConnectionWithImpl(SnowflakeConnectionV1.java:116)
[SF_KAFKA_CONNECTOR] net.snowflake.client.jdbc.SnowflakeConnectionV1.<init>(SnowflakeConnectionV1.java:96)
[SF_KAFKA_CONNECTOR] net.snowflake.client.jdbc.SnowflakeDriver.connect(SnowflakeDriver.java:172)
[SF_KAFKA_CONNECTOR] com.snowflake.kafka.connector.internal.SnowflakeConnectionServiceV1.<init>(SnowflakeConnectionServiceV1.java:69)
[SF_KAFKA_CONNECTOR] com.snowflake.kafka.connector.internal.SnowflakeConnectionServiceFactory$SnowflakeConnectionServiceBuilder.build(SnowflakeConnectionServiceFactory.java:75)
[SF_KAFKA_CONNECTOR] com.snowflake.kafka.connector.SnowflakeSinkConnector.validate(SnowflakeSinkConnector.java:234)
[SF_KAFKA_CONNECTOR] org.apache.kafka.connect.runtime.AbstractHerder.validateConnectorConfig(AbstractHerder.java:465)
[SF_KAFKA_CONNECTOR] org.apache.kafka.connect.runtime.AbstractHerder.lambda$validateConnectorConfig$2(AbstractHerder.java:365)
[SF_KAFKA_CONNECTOR] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[SF_KAFKA_CONNECTOR] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[SF_KAFKA_CONNECTOR] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[SF_KAFKA_CONNECTOR] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[SF_KAFKA_CONNECTOR] java.base/java.lang.Thread.run(Thread.java:829), errorCode:1001 (com.snowflake.kafka.connector.SnowflakeSinkConnector:236)
```

Is there any breaking changes introduced by this version ?

This is 100% reproducible, let me know if you need more details

**Comment by sfc-gh-japatel** [comment]
Hello, is your snowflake connection through private link?

**Comment by vdesabou** [comment]
@sfc-gh-japatel Not at all, it's using public internet. I'm deploying from my laptop using Docker. See full example [here](https://github.com/vdesabou/kafka-docker-playground/tree/master/connect/connect-snowflake-sink). Let me know how I can help for troubleshooting, it's 100% reproducible with my example

**Comment by sfc-gh-japatel** [comment]
Hi, can you check with 1.6.4? We have had couple of bugfixes in JDBC (KC dependency).

**Comment by sfc-gh-japatel** [comment]
Also, mind sharing the URL?

**Comment by vdesabou** [comment]
@sfc-gh-japatel 1.6.4 has fixed the problem, see https://github.com/vdesabou/kafka-docker-playground/issues/1899#issuecomment-1013319479

thanks !

**Comment by sfc-gh-japatel** [comment]
Hi @vdesabou thanks for the feedback. the link you shared is still showing v1.6.3. I am kind of confused if it was the test or version? 

Do we need to deprecate v1.6.3 in that case?

**Comment by vdesabou** [comment]
It is a bit misleading but the tests were actually ran with 1.6.4.
I believe 1.6.3 is broken so it would make sense to deprecate it

**Comment by sfc-gh-japatel** [comment]
Sounds good, what I am confused about is why my tests would pass in 1.6.3. If you could send me some logs, that would be great,. (I would want to look at JDBC logs because this seems connection issue)

But, I would deprecate this version at least in my release notes.

**Comment by vdesabou** [comment]
Sure I can do that on Monday. You can also easily run the example on your side, you just need Docker installed.

what kind of DEBUG traces you need ?

this ? https://github.com/vdesabou/kafka-docker-playground/blob/8b9654a60f0de2409c4b0822e388cd474bac28bb/connect/connect-snowflake-sink/docker-compose.plaintext.yml#L9

**Comment by vdesabou** [comment]
@sfc-gh-japatel Please find logs for 1.6.3 (not working) and 1.7.0 (working), again, this is exactly same test executed 

[logs.zip](https://github.com/snowflakedb/snowflake-kafka-connector/files/7880558/logs.zip)

hope that helps

**Comment by sfc-gh-japatel** [comment]
Thank you @vdesabou Confirmed that we had issues with _ in account name. So we fixed that in JDBC 3.13.13 (hence 1.7.0)
